crates/engine-primitives/src/lib.rs

```diff
@@ -64,312 +58,312 @@ pub trait EngineTypes:
-///
-/// This is used to ensure that the correct error code is returned when validating the payload or
-/// payload attributes.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum MessageValidationKind {
-    /// We are validating fields of a payload attributes.
-    PayloadAttributes,
-    /// We are validating fields of a payload.
-    Payload,
-}
-
-impl MessageValidationKind {
-    /// Returns an `EngineObjectValidationError` based on the given
-    /// `VersionSpecificValidationError` and the current validation kind.
-    pub const fn to_error(
-        self,
-        error: VersionSpecificValidationError,
-    ) -> EngineObjectValidationError {
-        match self {
-            Self::Payload => EngineObjectValidationError::Payload(error),
-            Self::PayloadAttributes => EngineObjectValidationError::PayloadAttributes(error),
-        }
-    }
-}
-
-/// Validates the presence or exclusion of fork-specific fields based on the ethereum execution
-/// payload, or payload attributes, and the message version.
-///
-/// The object being validated is provided by the [PayloadOrAttributes] argument, which can be
-/// either an execution payload, or payload attributes.
-///
-/// The version is provided by the [EngineApiMessageVersion] argument.
-pub fn validate_version_specific_fields<Type>(
-    chain_spec: &ChainSpec,
-    version: EngineApiMessageVersion,
-    payload_or_attrs: PayloadOrAttributes<'_, Type>,
-) -> Result<(), EngineObjectValidationError>
-where
-    Type: PayloadAttributes,
-{
-    validate_withdrawals_presence(
-        chain_spec,
-        version,
-        payload_or_attrs.message_validation_kind(),
-        payload_or_attrs.timestamp(),
-        payload_or_attrs.withdrawals().is_some(),
-    )?;
-    validate_parent_beacon_block_root_presence(
-        chain_spec,
-        version,
-        payload_or_attrs.message_validation_kind(),
-        payload_or_attrs.timestamp(),
-        payload_or_attrs.parent_beacon_block_root().is_some(),
-    )
-}
-
-/// The version of Engine API message.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
-pub enum EngineApiMessageVersion {
-    /// Version 1
-    V1,
-    /// Version 2
-    ///
-    /// Added in the Shanghai hardfork.
-    V2,
-    /// Version 3
-    ///
-    /// Added in the Cancun hardfork.
-    V3,
-    /// Version 4
-    ///
-    /// Added in the Prague hardfork.
-    V4,
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn version_ord() {
-        assert!(EngineApiMessageVersion::V4 > EngineApiMessageVersion::V3);
-    }
-}
+// /// Validates the timestamp depending on the version called:
+// ///
+// /// * If V2, this ensures that the payload timestamp is pre-Cancun.
+// /// * If V3, this ensures that the payload timestamp is within the Cancun timestamp.
+// /// * If V4, this ensures that the payload timestamp is within the Prague timestamp.
```

Comment:
can be removed?

Reply:
yes, I'm working on another branch. based on current changes, will force push to this one or create another PR. Thanks for your feedback.